components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -734,14 +765,15 @@ public User updateUser(User user, Map<String, Boolean> requiredAttributes) throw
     /**
      * Method to decide whether to use new APIs or legacy APIs.
      *
-     * @param limit Number of results required for the filter request.
+     * @param limit Number of results required for the filter request (limit equals to ZERO will retrieve all users
+     *              who matches the filter).
      * @return True if legacy API filtering is needed.
      */
-    private boolean isUseLegacyAPIs(int limit) {
+    private boolean isUseLegacyAPIs(Integer limit) {
 
         // If the limit is not specified, list all the users using old APIs since the new APIs must have a
         // limit larger than zero.
-        if (limit <= 0) {
+        if (limit == 0) {
```

Comment:
I don't see a necessity to change the signature

Reply:
fixed: 8f8c20292c4fac76778a8aa95490675023d7d063